src/internal/helper.ts

```diff
@@ -0,0 +1,588 @@
+    '"': '',
+    '&quot;': '',
+    '&#34;': '',
+    '&QUOT;': '',
+    '&#x00022': '',
+  }
+  return etag.replace(/^("|&quot;|&#34;)|("|&quot;|&#34;)$/g, (m) => replaceChars[m] as string)
+}
+
+function objectToBuffer(payload: Binary): Buffer {
+  // don't know how to write this...
+  return Buffer.from(payload)
+}
+
+export function toMd5(payload: Binary): string {
+  let payLoadBuf: Binary = objectToBuffer(payload)
+  // use string from browser and buffer from nodejs
+  // browser support is tested only against minio server
+  payLoadBuf = isBrowser ? payLoadBuf.toString() : payLoadBuf
+  return crypto.createHash('md5').update(payLoadBuf).digest().toString('base64')
+}
+
+export function toSha256(payload: Binary): string {
+  return crypto.createHash('sha256').update(payload).digest('hex')
+}
+
+/**
+ * toArray returns a single element array with param being the element,
+ * if param is just a string, and returns 'param' back if it is an array
+ * So, it makes sure param is always an array
+ */
+export function toArray<T = unknown>(param: T | T[]): Array<T> {
+  if (!Array.isArray(param)) {
+    return [param] as T[]
+  }
+  return param
+}
+
+export function sanitizeObjectKey(objectName: string): string {
+  // + symbol characters are not decoded as spaces in JS. so replace them first and decode to get the correct result.
+  const asStrName = (objectName ? objectName.toString() : '').replace(/\+/g, ' ')
+  return decodeURIComponent(asStrName)
+}
+
+export const PART_CONSTRAINTS = {
+  // absMinPartSize - absolute minimum part size (5 MiB)
+  ABS_MIN_PART_SIZE: 1024 * 1024 * 5,
+  // MIN_PART_SIZE - minimum part size 16MiB per object after which
+  MIN_PART_SIZE: 1024 * 1024 * 16,
+  // MAX_PARTS_COUNT - maximum number of parts for a single multipart session.
+  MAX_PARTS_COUNT: 10000,
+  // MAX_PART_SIZE - maximum part size 5GiB for a single multipart upload
+  // operation.
+  MAX_PART_SIZE: 1024 * 1024 * 1024 * 5,
+  // MAX_SINGLE_PUT_OBJECT_SIZE - maximum size 5GiB of object per PUT
+  // operation.
+  MAX_SINGLE_PUT_OBJECT_SIZE: 1024 * 1024 * 1024 * 5,
+  // MAX_MULTIPART_PUT_OBJECT_SIZE - maximum size 5TiB of object for
+  // Multipart operation.
+  MAX_MULTIPART_PUT_OBJECT_SIZE: 1024 * 1024 * 1024 * 1024 * 5,
+}
+
+const GENERIC_SSE_HEADER = 'X-Amz-Server-Side-Encryption'
+
+const ENCRYPTION_HEADERS = {
+  // sseGenericHeader is the AWS SSE header used for SSE-S3 and SSE-KMS.
+  sseGenericHeader: GENERIC_SSE_HEADER,
+  // sseKmsKeyID is the AWS SSE-KMS key id.
+  sseKmsKeyID: GENERIC_SSE_HEADER + '-Aws-Kms-Key-Id',
+} as const
+
+/**
+ * Return Encryption headers
+ * @param encConfig
+ * @returns an object with key value pairs that can be used in headers.
+ */
+export function getEncryptionHeaders(encConfig: Encryption): Header {
+  const encType = encConfig.type
+
+  if (!isEmpty(encType)) {
+    if (encType === ENCRYPTION_TYPES.SSEC) {
+      return {
+        [ENCRYPTION_HEADERS.sseGenericHeader]: 'AES256',
```

Comment:
https://github.com/minio/minio-js/blob/3ec0a688652fe36678376cc42bff834783019d36/src/helpers.js#L497-L500

I think it's a bug here